cmd/keymasterd/app.go

```diff
@@ -767,6 +763,20 @@ func (state *RuntimeState) getRequiredWebUIAuthLevel() int {
 	return AuthLevel
 }
 
+func (state *RuntimeState) reprocessUsername(username string) string {
```

Comment:
you have now a good entry point to perform testing. Can you please add some tests, in particular that your regular expression does what you want? ex:
foo@exampe@example.bar@example.com -> should this be "foo" OR foo@exampe@example.bar -> explain why.